src/hotspot/share/utilities/resourceHash.hpp

```diff
@@ -136,6 +138,25 @@ class ResourceHashtableBase : public STORAGE {
     }
   }
 
+ /**
+  * Inserts a value in the front of the table.
+  * Since the table is locked, this is ok.
+  * @return: true: a new item is always added
+  */
+  bool put_fast(K const& key, V const& value) {
+    unsigned hv = HASH(key);
```

Comment:
I think `put_fast` is not clear enough. Maybe `put_must_be_absent()` or something more concise.

Reply:
I would suggest `put_when_absent` to complement `put_if_absent` - with suitable descriptive comments of course.

Reply:
I cannot tell the difference between `put_when_absent` and `put_if_absent`. Grammatically they mean the same thing to me.

Reply:
My preference is to eventually make 'put' be 'put-ifwhen-absent', so I don't care which name you two pick.

Reply:
`put_when_known_absent`? 

A basic `put` should either add or replace; a `put_if_absent` should only add else do nothing.

Reply:
put_when_absent is what I have and it's fine.  I don't think we need more sentence names or changing doesn't materially improve this patch.  I was comparing to the std::unordered_map class which we want to minimally emulate and insert does insert if absent, so we shouldn't rewrite "put" to mean put_if/when_absent, but the existing behavior was surprising and unexpected to me.

https://en.cppreference.com/w/cpp/container/unordered_map/insert

Reply:
Well ... many of our API's are more Java oriented in naming rather than C++ containers. And unordered_map does not strike me as something we even want to minimally emulate when it comes to method naming. YMMV.